editor/src/messages/tool/transform_layer/transform_layer_message_handler.rs

```diff
@@ -908,4 +908,99 @@ mod test_transform_layer {
 		let translation_diff = (after_cancel.translation - original_transform.translation).length();
 		assert!(translation_diff < 1.0, "Translation component changed too much: {}", translation_diff);
 	}
+
+	#[tokio::test]
+	async fn test_transform_with_panned_view() {
+		let mut editor = EditorTestUtils::create();
+		editor.new_document().await;
+		editor.drag_tool(ToolType::Rectangle, 0., 0., 100., 100., ModifierKeys::empty()).await;
+		let document = editor.active_document();
+		let layer = document.metadata().all_layers().next().unwrap();
+		let original_transform = get_layer_transform(&mut editor, layer).await.unwrap();
+
+		let pan_amount = DVec2::new(200.0, 150.0);
+		editor.handle_message(NavigationMessage::CanvasPan { delta: pan_amount }).await;
+
+		// Applying a direct transform change in viewport space
+		editor
+			.handle_message(GraphOperationMessage::TransformChange {
+				layer,
+				transform: DAffine2::from_translation(DVec2::new(50.0, 50.0)),
+				transform_in: TransformIn::Viewport,
+				skip_rerender: false,
+			})
+			.await;
+
+		let final_transform = get_layer_transform(&mut editor, layer).await.unwrap();
+		let translation_delta = final_transform.translation - original_transform.translation;
```

Comment:
Panning the canvas doesn't impact a translation. I'd suggest you do a scale transformation.